### DISCUSSION TITLE
Welcome to DDDplus Discussions!

### DISCUSSION THREAD

**Discussion by funkygao** [funkygao/cp-ddd-framework#43] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by funkygao** [comment]
Hello world!

**Comment by Kongwei-Liao** [comment]
hello

**Comment by KuangShaoK** [comment]
作者进行过系统的思考，非常喜欢这里扩展点和工作流的理念，和SOFA的思想很像。我把dddplus-demo和ddd-framwork1.1看了一遍大致理解了，但是这个新的2.0版本和demo又看懵了，看起来十分原教旨，对以前的扩展点和Step做了些改动。能大致理解Router Policy的理念 但是这个demo: CheckingAppService 在框架基本用法没理解的情况下，加上混入了业务逻辑(感觉没有1.1清晰)，能看懂逻辑但属实难懂框架的正确用法。希望作者能写一些框架用法的文章，感谢。

**Reply by KuangShaoK** [reply]
我已经看了非常多ddd的文章，但还是不大理解2.0想怎么实现业务多态。起初是从这篇文章了解到的，https://juejin.cn/post/6956572939936350244。要是作者能结合demo片段大致讲一下各个扩展点的设计理念和落地就好了。持续关注中。

**Reply by funkygao** [reply]
1.0版本核心是扩展点机制，2.0版本是在扩展点基础上增加了DDD的builing blocks，并通过分析AST结合DSL来逆向生成业务模型。

CheckingAppService演示的内容没有像1.0版本的把扩展点作为重点，而是讲了2件事：
- 如何使用DDDplus的building blocks，来实现充血模型
- 如何进行DSL代码标注，这样就可以通过dddplus-maven-plugin自动生成多种业务模型视图

由于扩展点机制只是微调，没有做根本性改动，因此2.0版本里没有作为重点。

其中的Step修改内容deprecated，原因是业务逻辑编排，属于远离业务模型层的应用层，它与业务模型无关；此外，市场上有很多更好的选择，DDDplus的step没有突出优势，也不打算深入发展

**Reply by funkygao** [reply]
@KuangShaoK

**Reply by KuangShaoK** [reply]
感谢解惑，大概理解了2.0的重点是视图生成，看样子1.1除了step以外的机制基本都可以融合到2.0里，让我仔细研究一下。
